prow/logrusutil/logrusutil.go

```diff
@@ -73,3 +77,41 @@ func (f *DefaultFieldsFormatter) Format(entry *logrus.Entry) ([]byte, error) {
 		Caller:  entry.Caller,
 	})
 }
+
+// CensoringFormatter represents a logrus formatter that
+// can be used to censor sensitive information
+type CensoringFormatter struct {
+	delegate   logrus.Formatter
+	getSecrets func() sets.String
+}
+
+func (f CensoringFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+	raw, err := f.delegate.Format(entry)
+	if err != nil {
+		return raw, err
+	}
+	return f.Censor(raw), nil
+}
+
+const censored = "CENSORED"
+
+var censoredBytes = []byte(censored)
+
+// Censor replaces sensitive parts of the content with a placeholder.
+func (f CensoringFormatter) Censor(content []byte) []byte {
```

Comment:
This doesn't need to be exposed, does it? If anything this signature (`func([]byte) []byte`) should be injected behavior that you use to create a censoring formatter, the formatter just knows how to call it during the logging call.

Reply:
It was private in my first try.
It is there only for backwards compatibility (there are places calling this function):

https://github.com/kubernetes/test-infra/blob/7bc3b1341749328d76b26c8844099cdb2afe5b91/prow/config/secret/agent.go#L131-L138

With the formatter Censor function public, we can reuse the function:

https://github.com/kubernetes/test-infra/blob/9d3f3eac444b5284e7bc2fe278ffdc3e10af7424/prow/config/secret/agent.go#L118-L121

Removing `Censor` function in agent is another story: It is used to init github client. So i did not go that path very long. :)

Reply:
@stevekuznetsov 
Should I leave the agent's Censor function alone? Perhaps they (agent's and formatter's `Censor`) could have independent logic?

Reply:
I haven't investigated how exactly is agent's `Censor` function used, but Agent providing such function to censor any content sounds reasonable and potentially useful.

Maybe that logic could be a standalone function (not bound to the formatter), getting content and set of secrets to censor, and both agent and formatter would simply call it with their content?

Reply:
This is exactly how i did it in one of the commit. :)
On it now.

Reply:
Done.